controller/api/public/edges.go

```diff
@@ -96,14 +97,22 @@ func processEdgeMetrics(inbound, outbound model.Vector, resourceType, selectedNa
 	for _, sample := range inbound {
 		// skip inbound results without a clientID because we cannot construct edge
 		// information
-		if _, ok := sample.Metric[model.LabelName("client_id")]; ok {
-			key := sample.Metric[model.LabelName(resourceReplacementInbound)]
+		if clientID, ok := sample.Metric[model.LabelName("client_id")]; ok {
+			dstResource := string(sample.Metric[model.LabelName(resourceReplacementInbound)])
+
+			// format of clientId is id.namespace.serviceaccount.cluster.local
+			clientIDSlice := strings.Split(string(clientID), ".")
+			srcNs := clientIDSlice[1]
+			key := model.LabelValue(dstResource + srcNs)
 			dstIndex[key] = sample.Metric
 		}
 	}
 
 	for _, sample := range outbound {
-		key := sample.Metric[model.LabelName(resourceReplacementOutbound)]
+		dstResource := sample.Metric[model.LabelName(resourceReplacementOutbound)]
+		srcNs := sample.Metric[model.LabelName("namespace")]
+
+		key := dstResource + srcNs
```

Comment:
And then same comment here -- I'd use a "." separator. Also, the `key` var here is a string which is being coerced to a `model.LabelValue` when the map lookup happens, but for consistency with how the key is built above, I recommend explicitly typing the key a `model.LabelValue` as well.

```suggestion
		key := model.LabelValue(fmt.Sprintf("%s.%s", dstResource, srcNs))
```